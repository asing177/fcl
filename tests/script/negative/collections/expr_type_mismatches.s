map<int, text> foo = (1+1 : 15/3, 8*8 : "bar");
set<int> bar = {sender(), 99-99, 3.14};

transition initial -> terminal;

@initial
go() { transitionTo(@terminal); }
