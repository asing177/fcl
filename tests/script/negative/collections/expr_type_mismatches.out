Type error at Line 1, Column 48:
   Expected: 'text' inferred from literal '"bar"' on Line 1, Column 48
   But got:  'num' inferred from use of binary operator '/' on Line 1, Column 36

Error at Line 2, Column 23:
   Invalid definition for bar:
      Expected type: set<int>
      But inferred type: set<account> for expression '{3.14,(99 - 99),sender()}'

Type error at Line 2, Column 24:
   Expected: 'account' inferred from the return type of the function 'sender' on Line 2, Column 24
   But got:  'int' inferred from use of binary operator '-' on Line 2, Column 34

Error at Line 2, Column 36:
   Implicit loss of numeric precision
      'int' inferred from use of binary operator '-' on Line 2, Column 34
      'decimal<2>' inferred from literal '3.14' on Line 2, Column 41
   Possible fixes:
      • round the higher precision expression to match the lower precision
      • use types with more (e.g. arbitrary) precision